### ISSUE TITLE
Filtering Various Errors - WCAG 2.2/2.1/2.0 - ARIA 1.1/1.2

### ISSUE THREAD

**Issue by clapierre** [daisy/ace#422]
### Backstory: 
A third party has done some testing of a number of published EPUBs currently circulating.
This company validates all incoming EPUB content against the [WCAG AA 2.2 rules](https://linkprotect.cudasvc.com/url?a=https%3a%2f%2fdequeuniversity.com%2frules%2faxe%2f4.9&c=E,1,J1lbHifARImjogIX2FnXqrWyeV0HqVtu1ck90H6aytVtweIe-OnVD9TG0c4CVgqWC2R7OXoH6D31R0rnk68BiqLbIT7st2hlVN3yLhjkJ3kfnT0,&typo=1) using the [ACE validator](https://linkprotect.cudasvc.com/url?a=https%3a%2f%2fdaisy.org%2factivities%2fsoftware%2face%2f&c=E,1,lwWu6-Mf4WeO4G_fkeBlg3c-X6DabyV_nSUaaHfbHlYSo4f0jq_YhG66MNp5AJCfUHMX5VyzOmS9UdqECwiQp2_DrMyWWAlC1l6epGnB7OciavOoohbdPivalyrI&typo=1), developed by the DAISY consortium.

The reoccurring issues they found involve the rules: [aria-roles](https://linkprotect.cudasvc.com/url?a=https%3a%2f%2fdequeuniversity.com%2frules%2faxe%2f4.9%2faria-roles&c=E,1,EwD08wplhrR2Sb3KYO3v3UoDWicBNKbKPZLv9eppI6na2srM_Osbkl5tT2yhZuYCCOgbphnxielhoX2ZYjJAHWQj42sOkAPe_su_7LpKpc4,&typo=1) and [aria-required-children](https://linkprotect.cudasvc.com/url?a=https%3a%2f%2fdequeuniversity.com%2frules%2faxe%2f4.9%2faria-required-children&c=E,1,sOim-9dwoC04erQ0E9rKaP9oA9pYs-0ffrsRs3sfkwcKbUF0yldXF1Z3wBgMXl3Dk_i_7r38UyqTOAgdOk8L9F9VYozbIMuugvtwvCPHFqZ3uyOR-A,,&typo=1).

Also, most of the EPUBs that had failed did have an a11y_certifiedBy value of "Benetech", so it's unexpected that this title wouldn't have passed validation.

Also included in the metadata along with the GCA certifiedBy Benetech was also the conformance claim which in this case was the old IDPF link to the [Accessibility 1.0 specification which was WCAG 2.0 - AA](http://www.idpf.org/epub/a11y/accessibility-20160801.html#wcag-aa).

### Problem
With the EAA deadline looming, companies will be checking EPUB conformance using the latest version of ACE which may report errors that were not errors originally, with the previous version of ACE which was evaluating EPUB 3.2 files based on the WCAG 2.0 standard and the original DPUB ARIA rules.

We need a way to either switch between the various WCAG / ARIA rules or filter the results to hide those newer WCAG 2.1/2.2 & ARIA 1.2 errors.

### Potential Solution
1. provide a way to filter out newer errors or warnings via a command line option (This will allow folks to then set up which WCAG level they want to test against)

2. If a conformance claim is made within the EPUB's metadata which states this EPUB conformsTo (WCAG 2.0, 2.1, or 2.2) then automatically filter out any new WCAG / ARIA errors above the conformance level provided, so there is no confusion that this previously valid EPUB which was tested at the time to be fully conformant at a specific WCAG version is not flagged incorrectly because the standards has changed in the future.

This is extremally important and we need a solution quickly so that older EPUBs are not incorrectly flagged as having error and potentially rejected from ingestion because the latest version of ACE reported an error that didn't exist when the EPUB was initially produced using the older version of ACE.

**Comment by danielweck** [comment]
Hello Charles, Ace is being upgraded to the latest version of Deque Axe which is 4.10.0 at this moment in time. Initially we assumed that "mismatched" WCAG targets between EPUB Accessibility versions and depending on Ace/Axe revisions would be handled downstream by consumers of the JSON report generated by Ace (CLI), but this isn't a practical solution especially when using Ace App (GUI). I agree that a better solution would be to allow Ace users to filter ahead-of-time some Deque Axe rules/checks based on WCAG versions and levels. Regarding automatically inferring the filtering criteria based on authored metadata claims, this could be problematic in some cases, but this could be a sensible default as long as users can force the reporting of other WCAG versions / levels (essentially, there would be combination of blacklist / deny-list / block-list and whitelist / allow-list).

**Comment by danielweck** [comment]
Related: https://github.com/daisy/ace/issues/390
